curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/6610c6541c82551f95e765ab.md

```diff
@@ -0,0 +1,61 @@
+---
+id: 6610c6541c82551f95e765ab
+title: Step 51
+challengeType: 0
+dashedName: step-51
+---
+
+# --description--
+
+Variables can also be declared inside a function. These variables are considered to be in the <dfn>local scope</dfn>, or <dfn>block scope</dfn>. A variable declared inside a function can only be used inside that function. If you try to access it outside of the function, you will either get `undefined` or an error.
+
+To see this in action, use `const` to declare a `test` variable in your `padRow` function. Initialise it with the value `"Testing"`.
+
+Then, below your function, try to log `test` to the console. You will see an error because it is not defined outside of the function's local scope. Remove that `console.log` to pass the tests and continue.
+
+# --hints--
+
+Your function should declare a `test` variable.
+
+```js
+assert.match(padRow.toString(), /var\s+test/);
+```
+
+You should initialise `test` with the value `"Testing"`. Don't forget the semi-colon.
+
+```js
+assert.match(padRow.toString(), /var\s+test\s*=\s*('|")Testing\1;/)
```

Comment:
Because it `.toString()`s the function, which happens *after* Babel transforms the code. Oliver's got a PR to update the config, but at this moment parsing a function means testing for funky things like `var` instead of `let` or `const`.